les/client.go

```diff
@@ -74,8 +76,10 @@ type LightEthereum struct {
 	p2pServer *p2p.Server
 }
 
+type consensusCreatorFn func(stack *node.Node, chainConfig *params.ChainConfig, config *ethash.Config, notify []string, noverify bool, db ethdb.Database) consensus.Engine
+
 // New creates an instance of the light client.
-func New(stack *node.Node, config *eth.Config) (*LightEthereum, error) {
+func New(stack *node.Node, config *eth.Config, mkEngineFn consensusCreatorFn) (*LightEthereum, error) {
```

Comment:
It's unnecessary I think? We can move this function into the `consensus` package so that it can be shared.